src/Util.cpp

```diff
@@ -47,10 +47,25 @@ std::string get_env_variable(char const *env_var_name) {
     }
 
     #ifdef _MSC_VER
-    char lvl[128];
+    // call getenv_s without a buffer to determine the correct string length:
+    size_t length = 0;
+    if (getenv_s(&length, NULL, 0, env_var_name) != 0) {
```

Comment:
In both this case and the call below, I'd use an "||" operator to combine the error cases since the return the same thing. I'd probably also just `return ""` rather than doing `lvl.clear()` as having the return there indicates its an error out.